streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java

```diff
@@ -146,7 +146,8 @@
      *
      * @param reducer   a {@link Reducer} that computes a new aggregate result. Cannot be {@code null}.
      * @return a {@link KTable} that contains "update" records with unmodified keys, and values that represent the
-     * latest (rolling) aggregate for each key
+     * latest (rolling) aggregate for each key. Incase {@link KTable} returns null the following message will be
```

Comment:
typo: `In case`

nit: `null` -> `{@code null}`

Similar comments below.

What do you mean by `KTable returns null` ? Do you mean `Reducer#apply(..)` returns `null`?

Reply:
How about this: `If the reduce function returns null, it is then interpreted as deletion for the key, and future messages of the same key coming from upstream operators will be handled as newly initialized value`.

Reply:
Should we mention the "problem" with out-of-order data for this case? Should we ever recommend to _not_ return `null` ?

We had a discussion at some point to actually disallow returning `null` because a "delete" is not a valid aggregation result.

Reply:
It's a nuanced topic. It would be good to clarify if we can find a good way to say it.

The operator itself never processes data out of order. It always respects the order defined by the topic. But upstream repartitions don't provide any guarantees about the order they _write_ to the topic. This is how data can logically become out of order, even if users take care to populate their input topics strictly ordered, which might in turn violate some expectations with `null`s in the picture.

I'm concerned that if we start saying, "Kafka Streams processes data out of order", with no context, it'll create FUD. We should be up front about the limits of the system, but it doesn't benefit anyone to create the impression that we don't guarantee things that we actually do guarantee.

Reply:
In other words, I'm recommending that we specifically say something like "Producing deletes from your aggregations may cause unexpected results when processing dis-ordered data. Streams always processes data in the order it appears in the topic. If the topic is populated out of order, you may have late arriving records, which can cause records to become unexpectedly re-created after they have been deleted. Out-of-order data can be a problem for non-deleting aggregation functions as well, but it's especially surprising with aggregations that produce deletes."

:/ ... you see what I mean by saying that it's a nuanced topic.

Reply:
I agree it's nuanced. (Don't agree with all you say above -- omit details for now)

Seems we get out-of-scope for this PR... Maybe JavaDocs are not the right place to document this anyway?

Reply:
I'd agree that javadocs are not the best place to document out-of-ordering here.